R/checkSystemPackage.R

```diff
@@ -0,0 +1,65 @@
+#############################################################################
+#
+# XLConnect
+# Copyright (C) 2010-2020 Mirai Solutions GmbH
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+#############################################################################
+
+#############################################################################
+#
+# XLConnect Package Installation: find system packages by name, including
+# installed files, using Debian or Red Hat package managers
+# 
+# example: with 
+#
+# Author: Simon Poltier, Mirai Solutions GmbH
+#
+#############################################################################
+
+ checkSystemPackage <- function (debianpkgname, rpmpkgname, versionpattern) {
+  dpkg <- function(args) {
+    suppressWarnings(system2("dpkg", args, stdout = TRUE, stderr = TRUE))
+  }
+  rpm <- function(args) {
+    suppressWarnings(system2("rpm", args, stdout = TRUE, stderr = TRUE))
+  }
+  distro <- function(distroName) {
```

Comment:
Could we add other Linux distros to the CI build matrix? E.g. RHEL, Fedora and Debian?

Reply:
Can try, but it doesn't come as part of GH actions hosted environments. It's possible to use docker containers, but some open source actions we use (e.g. setting up Java, setting up R.) might not support other distros.

Reply:
As expected, using containers we can't use the recommended actions for setting up java, R, and existing docker images are missing some of the dependencies we need (rocker - debian based - doesn't have java, there's an unmaintained centos7 + R image which probably we shouldn't use anyway).
See e.g. https://github.com/miraisolutions/xlconnect/actions/runs/536526701 and other actions failing on the same day.

My attempts at making GH actions work with containers are on branch https://github.com/miraisolutions/xlconnect/tree/custom/gh-actions-container.

I can start working on a complete centos / red hat UBI / debian docker image for testing XLConnect, but I'm not sure if that should be part of this PR.
I have of course tested the package resolution manually on centos.

Reply:
Following up with #121